js/apps/admin-ui/src/client-scopes/details/ScopeForm.tsx

```diff
@@ -19,29 +22,93 @@ import {
 import { useRealm } from "../../context/realm-context/RealmContext";
-import { useLoginProviders } from "../../context/server-info/ServerInfoProvider";
+import {
+  useLoginProviders,
+  useServerInfo,
+} from "../../context/server-info/ServerInfoProvider";
 import { convertAttributeNameToForm, convertToFormValues } from "../../util";
 import useIsFeatureEnabled, { Feature } from "../../utils/useIsFeatureEnabled";
 import { toClientScopes } from "../routes/ClientScopes";
 
 const OID4VC_PROTOCOL = "oid4vc";
 
+// Validation function for comma-separated lists
+const validateCommaSeparatedList = (value: string | undefined) => {
+  if (!value || value.trim() === "") {
+    return true;
+  }
+  if (value.includes(", ") || value.includes(" ,")) {
+    return "Comma-separated list must not contain spaces around commas";
+  }
+  const entries = value.split(",");
+  const hasEmptyEntries = entries.some((entry) => entry.trim() === "");
+  if (hasEmptyEntries) {
+    return "Comma-separated list contains empty entries";
+  }
+  return true;
+};
+
 type ScopeFormProps = {
   clientScope?: ClientScopeRepresentation;
   save: (clientScope: ClientScopeDefaultOptionalType) => void;
 };
 
 export const ScopeForm = ({ clientScope, save }: ScopeFormProps) => {
   const { t } = useTranslation();
+  const { adminClient } = useAdminClient();
   const form = useForm<ClientScopeDefaultOptionalType>({ mode: "onChange" });
   const { control, handleSubmit, setValue, formState } = form;
   const { isDirty, isValid } = formState;
   const { realm } = useRealm();
 
   const providers = useLoginProviders();
+  const serverInfo = useServerInfo();
   const isFeatureEnabled = useIsFeatureEnabled();
   const isDynamicScopesEnabled = isFeatureEnabled(Feature.DynamicScopes);
 
+  // Get available signature algorithms from server info
+  const signatureAlgorithms = useMemo(
+    () =>
+      serverInfo?.providers?.signature?.providers
+        ? Object.keys(serverInfo.providers.signature.providers)
+        : [],
+    [serverInfo],
+  );
+
+  // Fetch realm keys for signing_key_id dropdown
+  const [realmKeys, setRealmKeys] = useState<KeyMetadataRepresentation[]>([]);
+
+  useFetch(
+    async () => {
+      const keysMetadata = await adminClient.realms.getKeys({ realm });
+      return keysMetadata.keys || [];
+    },
+    setRealmKeys,
+    [],
+  );
+
+  // Prepare key options for SelectControl
+  // Filter only active keys suitable for signing credentials
+  const keyOptions = useMemo(() => {
+    const options = [{ key: "", value: t("useDefaultKey") }];
+    if (realmKeys && realmKeys.length > 0) {
+      const keyOptions = realmKeys
+        .filter(
+          (key) =>
+            key.kid &&
+            key.status === "ACTIVE" &&
+            key.algorithm &&
+            signatureAlgorithms.includes(key.algorithm),
+        )
+        .map((key) => ({
+          key: key.kid!,
+          value: `${key.kid}${key.algorithm ? ` (${key.algorithm})` : ""}`,
```

Comment:
There is already a filter on the algorithm.
```suggestion
          value: `${key.kid} (${key.algorithm})`,
```